egs2/TEMPLATE/spk1/spk.sh

```diff
@@ -0,0 +1,262 @@
+fi
+
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    if "${skip_train}"; then
+        if "${eval_valid_set}"; then
+            _dsets="${valid_set} ${test_sets}"
+        else
+            _dsets="${test_sets}"
+        fi
+    else
+        _dsets="${train_set} ${valid_set} ${test_sets}"
+    fi
+
+    if [ "${feats_type}" = raw ]; then
+        log "Stage 2: Format wav.scp: data/ -> ${data_feats}"
+        for dset in ${_dsets}; do
+            utils/copy_data_dir.sh --validate_opts --non-print data/"${dset}" "${data_feats}/${dset}"
+            echo "${feats_type}" > "${data_feats}$/${dset}/feats_type"
+
+            # shellcheck disable=SC2086
+            scripts/audio/format_wav_scp.sh --nj "${nj}" --cmd "${train_cmd}" \
+                --audio-format "${audio_format}" --fs "${fs}" \
+                --multi-columns-input "${multi_columns_input_wav_scp}" \
+                --multi-columns-output "${multi_columns_output_wav_scp}" \
+                "data/${dset}"/wav.scp" "${data_feats}"/${dset}"
+
+        done
+    else
+        log "${feats_type} is not supported yet."
+        exit 1
+    fi
+fi
+
+if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
+    echo "Stage 3: Collect stats"
+    _spk_train_dir="${data_feats}/${train_set}"
+    _spk_valid_dir="${data_feats}/${valid_set}"
+
+    if [ -n "${spk_config}"  ]; then
+        # To generate the config file: e.g.
+        #   % python3 -m espnet2.bin.spk_train --print_config --optim adam
+        _opts+="--config ${spk_config} "
+    fi
+
+    _scp=wav.scp
+    if [[ "${audio_format}" == *ark* ]]; then
+        _type=kaldi_ark
+    else
+        # sound supports "wav", "flac", etc.
+        _type=sound
+    fi
+
+    # 1. Split key file
+    _logdir="${spk_stats_dir}/logdir"
+    mkdir -p "${_logdir}"
+
+    _nj=$(min "${nj}" "$(<${_spk_train_dir}/${_scp} wc -l)" "$(<${_spk_valid_dir}/${_scp} wc -l)")
+
+    key_file="${_spk_train_dir}/${_scp}"
+    split_scps=""
+    for n in $(seq "${_nj}"); do
+        split_scps+=" ${_logdir}/train.${n}.scp"
+    done
+    utils/split_scp.pl "${key_file}" ${split_scps}
+
+    key_file="${_spk_valid_dir}/${_scp}"
+    split_scps=""
+    for n in $(seq "${_nj}"); do
+        split_scps+=" ${_logdir}/valid.${n}.scp"
+    done
+    utils/split_scp.pl "${key_file}" ${split_scps}
+
+    # 2. Generate run.sh
+    log "Generate '${spk_stats_dir}/run.sh'. You can resume the process from stage 3 using this script"
+    mkdir -p "${spk_stats_dir}"; echo "${run_args} -- stage3 \"\$@\"; exit \$?" > "${spk_stats_dir}/run.sh"; chmod +x "${spk_stats_dir}/run.sh"
+
+    # 3. Submit jobs
+    log "Speaker collect-stats started... log: '${_logdir}/stats.*.log'"
+
+    # shellcheck disable=SC2046,SC2086
+    ${train_cmd} JOB=1:"${_nj}" "${_logdir}"/stats.JOB.log \
+        ${python} -m espnet2.bin.spk_train \
+            --collect_stats true \
+            --train_data_path_and_name_and_type ${_spk_train_dir}/wav.scp,speech,sound \
```

Comment:
You have defined the _type in line 165-170, it would be better to reuse it (for supporting ark file reading)